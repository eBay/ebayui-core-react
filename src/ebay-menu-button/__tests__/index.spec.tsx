import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import initStoryshots from '@storybook/addon-storyshots'
import { EbayMenuButton, EbayMenuButtonItem } from '../index'

initStoryshots({
    config: ({ configure }) => configure(() => require('./index.stories.tsx'), module)
})

const spy = jest.fn()

describe('<EbayMenuButton>', () => {
    describe('on button click', () => {
        it('should fire onExpand event', () => {
            const wrapper = render(
                <EbayMenuButton onExpand={spy}>
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )
            fireEvent.click(wrapper.container.querySelector('button'))

            expect(spy).toBeCalled()
        })
        it('should fire onCollapse event', () => {
            const wrapper = render(
                <EbayMenuButton onCollapse={spy}>
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            fireEvent.click(button)

            expect(spy).toBeCalled()
        })
    })

    describe('on menu item click', () => {
        it('should fire onSelect event', () => {
            const wrapper = render(
                <EbayMenuButton onSelect={spy}>
                    <EbayMenuButtonItem value="first" />
                    <EbayMenuButtonItem value="second" />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            const item = screen.getAllByRole('menuitem')[1]
            fireEvent.click(item)

           
            expect(spy).toBeCalled()
        })
    })

    describe('type="radio"', () => {
        it('should fire onChange event', () => {
            const wrapper = render(
                <EbayMenuButton type="radio" onChange={spy}>
                    <EbayMenuButtonItem value="first" />
                    <EbayMenuButtonItem value="second" />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            const item = screen.getAllByRole('menuitemradio')[1]
            fireEvent.click(item)

            expect(spy).toBeCalledWith()
        })
    })

    describe('type="checkbox"', () => {
        it('should fire onChange event', () => {
            const wrapper = render(
                <EbayMenuButton type="checkbox" onChange={spy}>
                    <EbayMenuButtonItem value="first" />
                    <EbayMenuButtonItem value="second" />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            const item = screen.getAllByRole('menuitemcheckbox')[1]
            fireEvent.click(item)

            

            expect(spy).toBeCalled()
        })
    })
})
