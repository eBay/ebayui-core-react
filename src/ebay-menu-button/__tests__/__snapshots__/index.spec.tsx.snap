// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-menu-button Borderless 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--borderless"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          eBay Menu without borders!
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Default 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          eBay Menu
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Disabled 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    disabled={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          eBay Menu
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Expanded 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={true}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          eBay Menu
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
  <div
    className="menu-button__menu menu"
    tabIndex={-1}
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span
          aria-hidden={false}
        >
          item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-16"
          />
        </svg>
      </div>
      <div
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <span
          aria-hidden={false}
        >
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-16"
          />
        </svg>
      </div>
      <div
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <span
          aria-hidden={false}
        >
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-16"
          />
        </svg>
      </div>
    </div>
  </div>
</span>
`;

exports[`Storyshots ebay-menu-button Fixed Width 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          Menu has a button width
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Multi-Select Menu Button 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          Filter
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Priorities 1`] = `
Array [
  <span
    className="menu-button"
  >
    <button
      a11y-text="Menu"
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn menu-button__button btn--primary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Primary
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
  <span
    className="menu-button"
  >
    <button
      a11y-text="Menu"
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn menu-button__button btn--tertiary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Tertiary
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
]
`;

exports[`Storyshots ebay-menu-button Reverse (Menu grows to the left) 1`] = `
<div
  style={
    Object {
      "marginLeft": "100px",
    }
  }
>
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Menu grows to the left
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>
</div>
`;

exports[`Storyshots ebay-menu-button Single-Select Menu Button (collapse on click) 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          Sort order: Price
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Single-Select Menu Button (item.checked) 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          Sort order
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Single-Select Menu Button (menu.checked) 1`] = `
Array [
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Topic: Phones
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
  " ",
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-live={null}
      className="btn menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Sort order: Relevance
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
]
`;

exports[`Storyshots ebay-menu-button Variants 1`] = `
Array [
  <h3>
    Button
  </h3>,
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu inside the form"
      aria-live={null}
      className="btn menu-button__button btn--secondary"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Button
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
  <h3>
    Form
  </h3>,
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu inside the form"
      aria-live={null}
      className="btn menu-button__button btn--form"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <span
        className="btn__cell"
      >
        <span
          className="btn__text"
        >
          <span>
            Form
          </span>
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--chevron-down-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-chevron-down-16"
          />
        </svg>
      </span>
    </button>
  </span>,
  <h3>
    Overflow
  </h3>,
  <span
    className="menu-button"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Menu"
      className="menu-button__button icon-btn"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="icon icon--overflow-vertical-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-overflow-vertical-16"
        />
      </svg>
    </button>
  </span>,
]
`;

exports[`Storyshots ebay-menu-button With Custom Label 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span
          style={
            Object {
              "background": "url(\\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='25' height='15' viewBox='0 0 5 3'%3E%3Cpath d='M0 0h5v3H0z'/%3E%3Cpath fill='%23D00' d='M0 1h5v2H0z'/%3E%3Cpath fill='%23FFCE00' d='M0 2h5v1H0z'/%3E%3C/svg%3E\\") no-repeat 50% 50%",
              "display": "inline-block",
              "height": "20px",
              "marginRight": "8px",
              "verticalAlign": "middle",
              "width": "26px",
            }
          }
        />
        Fun with flags!
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button With Separator 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span>
          Complex menu
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button With icon 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <svg
          aria-hidden={true}
          className="icon icon--settings-16"
          focusable={false}
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-settings-16"
          />
        </svg>
        <span>
          Settings
        </span>
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--chevron-down-16"
        focusable={false}
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-chevron-down-16"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-menu-button Without toggle icon 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span>
      Menu
    </span>
  </button>
</span>
`;
